fireant/queries/builder/dataset_blender_query_builder.py

```diff
@@ -58,56 +61,95 @@ def _find_dataset_fields_needed_to_be_mapped(dataset):
 
 
 @listify
-def _map_field(dataset, fields, dimension_map=None):
-    for field in fields:
+def _map_field(dataset_fields, blender_dataset_fields, dimension_map=None):
+    """
+    Returns a list of tuples shaped as blended dataset field and its matching field in the provided fields. Blender
+    fields that have to match in the dataset fields will be ignored. Please note that the Field class has a custom
+    dunder hash method.
+
+    :param dataset_fields: A set of Field instances pertaining to a single dataset (i.e. primary or secondary).
+    :param blender_dataset_fields: A list of Field instances in the blender dataset.
+    :param dimension_map: A dict mapping Field instances from different datasets.
+                          Often used to express the equivalency of primary and secondary dataset fields.
+    :return: A list of tuples shaped as blended dataset field and its matching field in the provided fields.
+    """
+    for field in blender_dataset_fields:
         if dimension_map is not None and field.definition in dimension_map:
             yield field, dimension_map[field.definition]
-        if field.definition in dataset.fields:
+        if field.definition in dataset_fields:
             yield field, field.definition
             # also yield the dataset field mapped to itself so that any reference to this field (from blender or
             # from dataset) can be used.
             yield field.definition, field.definition
-        if field in dataset.fields:
+        if field in dataset_fields:
             yield field, field
 
 
-def _datasets_and_field_maps(blender):
+def _datasets_and_field_maps(blender_dataset, filters):
+    """
+    Returns a list of tuples shaped as dataset and its fields mapped per the blender dataset ones. The field maps
+    are later used for knowing which columns to query on the respective primary and secondary datasets, given the
+    columns selected for the blender dataset.
+
+    :param blender: A DataSetBlender instance.
```

Comment:
You probably want to rename this parameter.

Reply:
Good catch. I'll do.

Reply:
Fixed.